runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -744,12 +744,17 @@ function M.get_virtual_text_chunks_for_line(bufnr, line, line_diags, opts)
   opts = opts or {}
   local prefix = opts.prefix or "■"
   local spacing = opts.spacing or 4
+  local severity_limit = opts.severity_limit
 
   -- Create a little more space between virtual text and contents
   local virt_texts = {{string.rep(" ", spacing)}}
 
   for i = 1, #line_diags - 1 do
-    table.insert(virt_texts, {prefix, virtual_text_highlight_map[line_diags[i].severity]})
+    if not severity_limit then
+      table.insert(virt_texts, {prefix, virtual_text_highlight_map[line_diags[i].severity]})
+    else
+      table.insert(virt_texts, {prefix, virtual_text_highlight_map[line_diags[i][severity_level]]})
```

Comment:
Is that a typo ?
```suggestion
      table.insert(virt_texts, {prefix, virtual_text_highlight_map[line_diags[i][severity_limit]]})
```

Reply:
Sorry for my sloppyness, I submitted pr in a hurry just now, now I have time